### ISSUE TITLE
Conditional content with spans not working in 1.3 and 1.4

### ISSUE THREAD

**Issue by knuesel** [quarto-dev/quarto-cli#6940]
### Bug description

Sometimes I define conditional content using spans, such as

```
The format is [HTML]{.content-visible when-format="html"}[PDF]{.content-visible when-format="pdf"}.
```

(A div would not work here as it would put the content in a separate paragraph.)

This used to work with quarto 1.2 but with 1.3 it fails silently (the conditional content is never shown) and with 1.4 it gives an error.

### Steps to reproduce

Render the above example with `quarto render file.qmd --to html` or `quarto render file.qmd --to pdf`.

### Expected behavior

Each output includes only the corresponding conditional content.

### Actual behavior

It works with quarto 1.2.335.

With quarto 1.3.450 the conditional content is missing from both outputs (they show "The format is .").

With quarto 1.4.376 I get the following error:

```
Error running filter /opt/quarto/share/filters/main.lua:
Inline, list of Inlines, or string expected, got Block
stack traceback:
	/opt/quarto/share/filters/main.lua:148: in function </opt/quarto/share/filters/main.lua:138>
	(...tail calls...)
	/opt/quarto/share/filters/main.lua:725: in local 'callback'
	/opt/quarto/share/filters/main.lua:738: in upvalue 'run_emulated_filter_chain'
	/opt/quarto/share/filters/main.lua:773: in function </opt/quarto/share/filters/main.lua:770>
```

### Your environment

- OS: Pop!_OS 22.04

### Quarto check output

```
Quarto 1.4.376
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.8: OK
      Dart Sass version 1.55.0: OK
      Deno version 1.33.4: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.376
      Path: /opt/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: Installation From Path
      Path: /usr/bin
      Version: 2021

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.12
      Path: /usr/bin/python3
      Jupyter: 5.3.0
      Kernels: python3, julia-1.8, octave, sos, julia-1.9, atk, sagemath

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.1.2
      Path: /usr/lib/R
      LibPaths:
        - /home/j/R/x86_64-pc-linux-gnu-library/4.1
        - /usr/local/lib/R/site-library
        - /usr/lib/R/site-library
        - /usr/lib/R/library
      knitr: 1.40
      rmarkdown: 2.17

[✓] Checking Knitr engine render......OK
```

**Comment by mcanouil** [comment]
Where did you see conditional content was supposed to work in spans?
I don't believe this was ever supported or highlighted, so to me it was a side effect.

From <https://quarto.org/docs/authoring/conditional.html#content-visible>:

> To make content visible only for a given format, create a div (`:::`) with the `.content-visible` class. For example, here we mark content as only visible in HTML:

**Comment by cscheid** [comment]
> (A div would not work here as it would put the content in a separate paragraph.)

A simple workaround for now is to reuse the paragraph; We want to improve this but it's not high-priority. Is there a reason this doesn't work for you?

**Comment by knuesel** [comment]
Ah sorry I stumbled on this while trying to render an old document with quarto 1.4 so I assumed a regression while it was never officially supported (by the way, it looks like the feature is still working for code blocks while that is also not documented).

Yes paragraph duplication works fine in my  actual use case. But of course it's annoying having to maintain two copies of the same thing, and the number of copies explodes in more complex cases (e.g. if the same paragraph has one part with 2 conditional cases, and another part with 3, you need to maintain 6 copies). 

From what I understand the problem is

* each custom node type should have a handler
* for parsing, handlers are looked up in a table using the class name

so we can't currently parse divs *and* spans using the same class name.

One solution that comes to mind would be to store handlers for different "kinds" in separate tables (one for Block handlers, one for Inline handlers). Would you be interested in a PR along these lines?

**Comment by cscheid** [comment]
> Yes paragraph duplication works fine in my actual use case. But of course it's annoying having to maintain two copies of the same thing, and the number of copies explodes in more complex cases (e.g. if the same paragraph has one part with 2 conditional cases, and another part with 3, you need to maintain 6 copies).

100% agreed. Again, this is a matter of available time, not of understanding the need :)

@knuesel In 1.3 that was the case, but in 1.4 we do have support for custom Inline nodes as well (we need it eg for handling shortcodes through custom AST nodes).

> Would you be interested in a PR along these lines?

We're always interested in PRs! Take a look at the [contributor guide](https://github.com/quarto-dev/quarto-cli/blob/main/CONTRIBUTING.md) and please go ahead!

**Comment by cwickham** [comment]
On the docs side, we do currently suggest you can use `.content-visible` in spans at the end of the [Projects > Profiles > Profile Content](https://quarto.org/docs/projects/profiles.html#profile-content).

I'm going to remove that paragraph since it isn't currently supported.

**Comment by knuesel** [comment]
Fixed by #7083.
